app/src/main/java/com/nextcloud/talk/adapters/messages/IncomingPollMessageViewHolder.kt

```diff
@@ -0,0 +1,291 @@
+    @JvmField
+    var ncApi: NcApi? = null
+
+    lateinit var message: ChatMessage
+
+    lateinit var reactionsInterface: ReactionsInterface
+
+    @SuppressLint("SetTextI18n")
+    override fun onBind(message: ChatMessage) {
+        super.onBind(message)
+        this.message = message
+        sharedApplication!!.componentApplication.inject(this)
+
+        setAvatarAndAuthorOnMessageItem(message)
+
+        colorizeMessageBubble(message)
+
+        itemView.isSelected = false
+        binding.messageTime.setTextColor(ResourcesCompat.getColor(context?.resources!!, R.color.warm_grey_four, null))
+
+        // parent message handling
+        setParentMessageDataOnMessageItem(message)
+
+        setPollPreview(message)
+
+        Reaction().showReactions(message, binding.reactions, binding.messageTime.context, false)
+        binding.reactions.reactionsEmojiWrapper.setOnClickListener {
+            reactionsInterface.onClickReactions(message)
+        }
+        binding.reactions.reactionsEmojiWrapper.setOnLongClickListener { l: View? ->
+            reactionsInterface.onLongClickReactions(message)
+            true
+        }
+    }
+
+    private fun setPollPreview(message: ChatMessage) {
+        var pollId: String? = null
+        var pollName: String? = null
+
+        if (message.messageParameters != null && message.messageParameters!!.size > 0) {
+            for (key in message.messageParameters!!.keys) {
+                val individualHashMap: Map<String?, String?> = message.messageParameters!![key]!!
+                if (individualHashMap["type"] == "talk-poll") {
+                    pollId = individualHashMap["id"]
+                    pollName = individualHashMap["name"].toString()
+                }
+            }
+        }
+
+        if (pollId != null && pollName != null) {
+            binding.messagePollTitle.text = pollName
+
+            val roomToken = (payload as? MessagePayload)!!.currentConversation.token!!
+            val isOwnerOrModerator = (payload as? MessagePayload)!!.currentConversation.isParticipantOwnerOrModerator
+
+            binding.bubble.setOnClickListener {
+                val pollVoteDialog = PollMainDialogFragment.newInstance(
+                    message.activeUser!!,
+                    roomToken,
+                    isOwnerOrModerator,
+                    pollId,
+                    pollName
+                )
+                pollVoteDialog.show(
+                    (binding.messagePollIcon.context as MainActivity).supportFragmentManager,
+                    TAG
+                )
+            }
+
+            val credentials = ApiUtils.getCredentials(message.activeUser?.username, message.activeUser?.token)
+            ncApi!!.getPoll(
+                credentials,
+                ApiUtils.getUrlForPoll(
+                    message.activeUser?.baseUrl,
+                    roomToken,
+                    pollId
+                )
+            ).subscribeOn(Schedulers.io())
```

Comment:
yep, I was aware of this. 
The MessageViewHolders in general need to be refactored. ideally while getting rid of chatkit at the same time..

Reply:
> yep, I was aware of this. The MessageViewHolders in general need to be refactored. ideally while getting rid of chatkit at the same time..

even if the rest of the viewholder can be refactored at a later time, this one should still use the `PollRepository` instead of `NCApi` directly

Reply:
it was decided in talk team to avoid so many requests. So i deleted this and now it's a static text..